From d323b6fd03449aeac1e4046be05518fa905595d4 Mon Sep 17 00:00:00 2001
From: Daniel Ammann <daniel.ammann@bytesatwork.ch>
Date: Tue, 5 Apr 2022 16:01:08 +0200
Subject: [PATCH] arm64: ti: k3-am625-bytedevkit: Add byteDEVKIT AM625x board
 support

This board is a carrier for the byteENGINE AM625x SOM.

Signed-off-by: Daniel Ammann <daniel.ammann@bytesatwork.ch>
---
 MAINTAINERS                                   |   1 +
 arch/arm64/boot/dts/ti/Makefile               |   1 +
 .../arm64/boot/dts/ti/k3-am625-bytedevkit.dts | 173 ++++++++++++++++++
 3 files changed, 175 insertions(+)
 create mode 100644 arch/arm64/boot/dts/ti/k3-am625-bytedevkit.dts

diff --git a/MAINTAINERS b/MAINTAINERS
index 803da1c28bf1..5e305b4b1313 100644
--- a/MAINTAINERS
+++ b/MAINTAINERS
@@ -4502,6 +4502,7 @@ M:	Daniel Ammann <daniel.ammann@bytesatwork.ch>
 S:	Maintained
 F:	.github/workflows/checkpatch.yml
 F:	Documentation/devicetree/bindings/arm/bytesatwork/bytesatwork.yaml
+F:	arch/arm64/boot/dts/ti/*bytedevkit*
 F:	arch/arm64/boot/dts/ti/*byteengine*
 
 C-MEDIA CMI8788 DRIVER
diff --git a/arch/arm64/boot/dts/ti/Makefile b/arch/arm64/boot/dts/ti/Makefile
index ed53580f2d4c..e045855c4cdc 100644
--- a/arch/arm64/boot/dts/ti/Makefile
+++ b/arch/arm64/boot/dts/ti/Makefile
@@ -13,6 +13,7 @@ dtb-$(CONFIG_ARCH_K3) += k3-am625-beagleplay.dtb
 dtb-$(CONFIG_ARCH_K3) += k3-am625-beagleplay-csi2-ov5640.dtbo
 dtb-$(CONFIG_ARCH_K3) += k3-am625-beagleplay-csi2-tevi-ov5640.dtbo
 dtb-$(CONFIG_ARCH_K3) += k3-am625-beagleplay-lincolntech-lcd185-panel.dtbo
+dtb-$(CONFIG_ARCH_K3) += k3-am625-bytedevkit.dtb
 dtb-$(CONFIG_ARCH_K3) += k3-am625-sk.dtb
 dtb-$(CONFIG_ARCH_K3) += k3-am625-sk-lincolntech-lcd185-panel.dtbo
 dtb-$(CONFIG_ARCH_K3) += k3-am625-sk-microtips-mf101hie-panel.dtbo
diff --git a/arch/arm64/boot/dts/ti/k3-am625-bytedevkit.dts b/arch/arm64/boot/dts/ti/k3-am625-bytedevkit.dts
new file mode 100644
index 000000000000..dba17befca1d
--- /dev/null
+++ b/arch/arm64/boot/dts/ti/k3-am625-bytedevkit.dts
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: GPL-2.0
+/*
+ * Copyright (C) 2022 bytes at work AG - https://www.bytesatwork.io
+ *
+ * based on k3-am625-sk.dts:
+ *
+ * Copyright (C) 2021-2022 Texas Instruments Incorporated - https://www.ti.com/
+ */
+
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include "k3-am625-byteengine.dtsi"
+
+/ {
+	compatible = "bytesatwork,bytedevkit-am62x", "bytesatwork,byteengine-am62x", "ti,am625";
+	model = "bytesatwork byteDEVKIT with byteENGINE AM62x";
+
+	aliases {
+		serial2 = &main_uart0;
+		mmc1 = &sdhci1;
+		usb0 = &usb0;
+		usb1 = &usb1;
+	};
+
+	chosen {
+		stdout-path = "serial2:115200n8";
+		bootargs = "console=ttyS2,115200n8 earlycon=ns16550a,mmio32,0x02800000";
+	};
+
+	extcon_usb1: extcon-usb1 {
+		compatible = "linux,extcon-usb-gpio";
+		id-gpios = <&main_gpio1 8 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&main_usb1_gpio_id_pins_default>;
+	};
+};
+
+&main_pmx0 {
+	main_uart0_pins_default: main-uart0-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x1c8, PIN_INPUT, 0) /* (D14) UART0_RXD */
+			AM62X_IOPAD(0x1cc, PIN_OUTPUT, 0) /* (E14) UART0_TXD */
+		>;
+	};
+
+	main_i2c1_pins_default: main-i2c1-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x1e8, PIN_INPUT_PULLUP, 0) /* (B17) I2C1_SCL */
+			AM62X_IOPAD(0x1ec, PIN_INPUT_PULLUP, 0) /* (A17) I2C1_SDA */
+		>;
+	};
+
+	main_mmc1_pins_default: main-mmc1-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x23c, PIN_INPUT, 0) /* (A21) MMC1_CMD */
+			AM62X_IOPAD(0x234, PIN_INPUT, 0) /* (B22) MMC1_CLK */
+			AM62X_IOPAD(0x230, PIN_INPUT, 0) /* (A22) MMC1_DAT0 */
+			AM62X_IOPAD(0x22c, PIN_INPUT, 0) /* (B21) MMC1_DAT1 */
+			AM62X_IOPAD(0x228, PIN_INPUT, 0) /* (C21) MMC1_DAT2 */
+			AM62X_IOPAD(0x224, PIN_INPUT, 0) /* (D22) MMC1_DAT3 */
+			AM62X_IOPAD(0x240, PIN_INPUT, 0) /* (D17) MMC1_SDCD */
+		>;
+	};
+
+	main_mdio1_pins_default: main-mdio1-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x160, PIN_OUTPUT, 0) /* (AD24) MDIO0_MDC */
+			AM62X_IOPAD(0x15c, PIN_INPUT, 0) /* (AB22) MDIO0_MDIO */
+		>;
+	};
+
+	main_rmii1_pins_default: main-rmii1-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x14c, PIN_INPUT, 1) /* (AB17) RMII1_RD0 */
+			AM62X_IOPAD(0x150, PIN_INPUT, 1) /* (AC17) RMII1_RD1 */
+			AM62X_IOPAD(0x148, PIN_INPUT, 1) /* (AD17) RMII1_REF_CLK */
+			AM62X_IOPAD(0x134, PIN_OUTPUT, 1) /* (AE20) RMII1_TD0 */
+			AM62X_IOPAD(0x138, PIN_OUTPUT, 1) /* (AD20) RMII1_TD1 */
+			AM62X_IOPAD(0x130, PIN_INPUT, 1) /* (AE19) RMII1_CRS_DV */
+			AM62X_IOPAD(0x12c, PIN_OUTPUT, 1) /* (AD19) RMII1_TX_EN */
+		>;
+	};
+
+	main_rmii2_pins_default: main-rmii2-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x184, PIN_INPUT, 1) /* (AE23) RMII2_RD0 */
+			AM62X_IOPAD(0x188, PIN_INPUT, 1) /* (AB20) RMII2_RD1 */
+			AM62X_IOPAD(0x180, PIN_INPUT, 1) /* (AD23) RMII2_REF_CLK */
+			AM62X_IOPAD(0x16c, PIN_OUTPUT, 1) /* (Y18) RMII2_TD0 */
+			AM62X_IOPAD(0x170, PIN_OUTPUT, 1) /* (AA18) RMII2_TD1 */
+			AM62X_IOPAD(0x168, PIN_INPUT, 1) /* (AE21) RMII2_CRS_DV */
+			AM62X_IOPAD(0x164, PIN_OUTPUT, 1) /* (AA19) RMII2_TX_EN */
+		>;
+	};
+
+	main_usb0_pins_default: main-usb0-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x0254, PIN_INPUT, 0) /* (C20) USB0_DRVVBUS */
+		>;
+	};
+
+	main_usb1_pins_default: main-usb1-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x0258, PIN_OUTPUT, 0) /* (F18) USB1_DRVVBUS */
+		>;
+	};
+
+	main_usb1_gpio_id_pins_default: main-usb1-gpio-id-pins-default {
+		pinctrl-single,pins = <
+			AM62X_IOPAD(0x0198, PIN_INPUT, 0) /* (A19) MCASP0_AXR2 */
+		>;
+	};
+};
+
+&main_uart0 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&main_uart0_pins_default>;
+};
+
+/* SD/MMC */
+&sdhci1 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&main_mmc1_pins_default>;
+	ti,driver-strength-ohm = <50>;
+	disable-wp;
+	no-1-8-v;
+};
+
+&usb0 {
+	dr_mode = "host";
+	pinctrl-names = "default";
+	pinctrl-0 = <&main_usb0_pins_default>;
+};
+
+&usbss1 {
+	ti,vbus-divider;
+};
+
+&usb1 {
+	dr_mode = "otg";
+	pinctrl-names = "default";
+	pinctrl-0 = <&main_usb1_pins_default>;
+	extcon = <&extcon_usb1>;
+	usb-role-switch;
+};
+
+&cpsw3g {
+	pinctrl-names = "default";
+	pinctrl-0 = <&main_mdio1_pins_default
+		     &main_rmii1_pins_default
+		     &main_rmii2_pins_default>;
+};
+
+&cpsw_port1 {
+	phy-mode = "rmii";
+	phy-handle = <&cpsw3g_phy0>;
+};
+
+&cpsw_port2 {
+	phy-mode = "rmii";
+	phy-handle = <&cpsw3g_phy1>;
+};
+
+&cpsw3g_mdio {
+	cpsw3g_phy0: ethernet-phy@0 {
+		reg = <0>;
+	};
+
+	cpsw3g_phy1: ethernet-phy@3 {
+		reg = <3>;
+	};
+};
-- 
2.39.2

